src/wrap.js

```diff
@@ -63,12 +63,15 @@ jQuery.fn.extend({
 		});
 	},
 
-	unwrap: function() {
-		return this.parent().each(function() {
-			if ( !jQuery.nodeName( this, "body" ) ) {
-				jQuery( this ).replaceWith( this.childNodes );
-			}
-		}).end();
+	unwrap: function( selector ) {
+		return this
+			.parent()
+				.filter( selector || ":not(body)" )
```

Comment:
We probably need to no-op on body children even when a selector is passed. I recommend `this.parent( selector ).not( "body" )`, and checking to see if storing it in a variable is smaller than this use of `.end()` to enable a one-liner.